### PR TITLE
[lcn] Make auto discovery use segment id and module id for ThingID

### LINKED ISSUE #8097: [lcn] Reference modules and groups by segment id and module/group id instead of serial number
I would like to propose to use a combination of segment and module/group id instead of the serial number to identify LCN things (when using automatic discovery).

It could be represented e.g. in the format 

**S**_<segment_id>_**M**_<module_id>_
resp.
**S**_<segment_id>_**G**_<group_id>_

LCN modules (smart/bus modules) are uniquely identified by this combination. Furthermore - as far as I know - using the module id (in combination with the segment id, if in bigger environments the installation is splitted into segments) is the common approach to identify modules when planning/programming LCN bus systems. (Logical) LCN groups are handled analogously.

In my opinion this would make textual item and rule definitions much more readable and maintainable.

### PR DESCRIPTION
**Pull request by toweosp**
When using auto discovery modules are now identified by a combination of segment id and module id in the format `S<segmentID>M<moduleID>` instead by serial number. Serial number is used as representation property (e.g. to identify identical modules when using auto-ignore option in PaperUI). Closes #8097.

### CODE REVIEW

**Comment by toweosp** [comment]
I intentionally used the serial/firmware number as the representation property and didn't remove the property. If I am not mistaken:

Serial number is the only unique and immutable property of an lcn module. So auto discovery can identify identity of modules/things even if segment id or module id get changed. Furthermore after the change provided by this PR auto discovery in existing installations using the 2.x lcn binding will not discover existing modules/things as new but ignore them if paper ui is configured accordingly. Moreover ThingID of manually added modules/things can differ from the scheme used by auto discovery, so I think ThingID will /can not be equal to the representation property in every case, might it be the serial number or an address identifier.

Last but not least I generally wouldn't remove the serial number property, because it could ease support in the not so rare cases where module handling depends on serial number/firmware age.

File: bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/LcnModuleDiscoveryService.java [hunk 1]
```diff
@@ -153,8 +153,10 @@ protected void startScan() {
 
                             ThingUID bridgeUid = localBridgeHandler.getThing().getUID();
                             String serialNumber = matcher.group("sn");
-                            ThingUID thingUid = new ThingUID(LcnBindingConstants.THING_TYPE_MODULE, bridgeUid,
-                                    serialNumber);
+
+                            String thingID = String.format("S%dM%d", addr.getSegmentId(), addr.getModuleId());
```
Comment (fwolter) [inline bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/LcnModuleDiscoveryService.java:157]: Add leading zeros to achieve sortable Thing IDs.
```suggestion
                            String thingID = String.format("S%03dM%03d", addr.getSegmentId(), addr.getModuleId());
```

**Comment by fwolter** [comment]
Can you change the Thing UIDs in the documentation examples from serial numbers to the new scheme?

**Comment by toweosp** [comment]
Done.

**Pull request merged**